### PR TITLE
Fix migration, merge conflicts, and ClientView link changes

### PR DESCRIPTION
**Pull request by gigxz**
Patching a bug I missed from https://github.com/greenriver/hmis-warehouse/pull/3009 that cause an error when trying to resolve the clientview link on the GQL Client.

### CODE REVIEW

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb [hunk 1]
```diff
@@ -4,13 +4,22 @@
...
 
+# To create a ClientView URL:
+# ::GrdaWarehouse::RemoteCredential.create!(
+# 	slug: 'ac_hmis_clientview',
+# 	endpoint: 'https://www.google.com', # URL base of ClientView
+# 	type: 'GrdaWarehouse::RemoteCredential',
+# 	username: '',
+# 	encrypted_password: '',
+# 	active: true,
+# )
```
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb]: This is what I did locally to store a base URL for client view. Is this correct with howit's supposed to be used?  @eanders @ttoomey

File: db/warehouse/migrate/20230428210859_clean_up_external_id_index.rb [hunk 2]
```diff
@@ -2,12 +2,17 @@ class CleanUpExternalIdIndex < ActiveRecord::Migration[6.1]
   # remove redundant indexes. These columns already indexed by external_ids_uniq_source_value
   # (source_id, source_type, remote_credential_id)
   def change
-    remove_index :external_ids,
-                 [:source_id, :source_type, :remote_credential_id, :value],
-                 unique: true,
-                 name: 'external_ids_uniq'
```
Comment (gigxz) [inline db/warehouse/migrate/20230428210859_clean_up_external_id_index.rb:8]: I didn't have this index in my db, and the tests were complaining too: https://github.com/greenriver/hmis-warehouse/actions/runs/4893839444/jobs/8737266921
Not sure why that happened, but I made it conditional.

Comment (ttoomey) [reply db/warehouse/migrate/20230428210859_clean_up_external_id_index.rb:8]: I assumed `index_external_ids_on_source` was created by rails automatically but it must just be cruft that ended up in the schema. Thanks for fixing it!

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb (hunk 1 above)
Comment (ttoomey) [reply drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb]: Good question. I was a little unclear on the clientview credential. It seems like it just exists to provide this one link_base variable but I'm probably missing something. Before `185000178-remote-credential-adjustments`,  what used to be there was:
```ruby
GrdaWarehouse::RemoteCredentials::ExternalLink.where(slug: 'clientview').first&.link_base
```

If you already had the clientview credential, it should have been [updated by this migration](https://github.com/greenriver/hmis-warehouse/pull/3009/files#diff-578626979af2debd8aff4977d01815b2ca6e9d22f777c92a8368dd46d50f5d67)

But to create it, I think what you have is fine, you just need to change the type to be ExternalLink (see suggestion above)

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb [hunk 3]
```diff
@@ -4,13 +4,22 @@
...
 ###
 
+# To create a ClientView URL:
+# ::GrdaWarehouse::RemoteCredential.create!(
+# 	slug: 'ac_hmis_clientview',
+# 	endpoint: 'https://www.google.com', # URL base of ClientView
+# 	type: 'GrdaWarehouse::RemoteCredential',
+# 	username: '',
+# 	encrypted_password: '',
+# 	active: true,
```
Comment (ttoomey) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb]:
```suggestion
# GrdaWarehouse::RemoteCredentials::ExternalLink.create!(
# 	slug: HmisExternalApis::AcHmis::Clientview::SYSTEM_ID,
# 	endpoint: 'https://www.google.com', # URL base of ClientView
# 	username: '',
# 	encrypted_password: '',
# 	active: true,
# )
```

**Review by ttoomey — APPROVED** [review]

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/clientview.rb]: I see, thanks. Yeah that's exactly what its for, just providing one URL value. Thanks for the suggestion, I figured I was missing a type. I'll update

**Comment by gigxz** [comment]
The previous commit was green, I'm going to go ahead and merge to get release branch back to a good state

**Pull request merged**